dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/put-fee-record-correction-transient-form-data.controller/index.ts

```diff
@@ -0,0 +1,66 @@
+import { ApiError, FeeRecordCorrectionTransientFormDataEntity, RecordCorrectionTransientFormData, REQUEST_PLATFORM_TYPE } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { CustomExpressRequest } from '../../../../../types/custom-express-request';
+import { NotFoundError } from '../../../../../errors';
+import { FeeRecordCorrectionRepo } from '../../../../../repositories/fee-record-correction-repo';
+import { FeeRecordCorrectionTransientFormDataRepo } from '../../../../../repositories/fee-record-correction-transient-form-data-repo';
+import { PutFeeRecordCorrectionTransientFormDataPayload } from '../../../../routes/middleware/payload-validation';
+
+export type PutFeeRecordCorrectionTransientFormDataRequest = CustomExpressRequest<{
+  params: {
+    bankId: string;
+    correctionId: string;
+  };
+  reqBody: PutFeeRecordCorrectionTransientFormDataPayload;
+}>;
+
+/**
+ * Controller for the PUT fee record correction transient form data route.
+ *
+ * Validates the correction exists and belongs to a report for the correct bank.
+ *
+ * Then creates a new fee record correction transient form data entity with the
+ * correction id, user id and form data, and saves it.
+ * @param req - The {@link PutFeeRecordCorrectionTransientFormDataRequest} request object
+ * @param res - The response object
+ */
+export const putFeeRecordCorrectionTransientFormData = async (req: PutFeeRecordCorrectionTransientFormDataRequest, res: Response) => {
+  try {
+    const { correctionId: correctionIdString, bankId } = req.params;
+    const { user, formData } = req.body;
+
+    const correctionId = Number(correctionIdString);
+    const userId = user.id;
+
+    const correction = await FeeRecordCorrectionRepo.findByIdAndBankId(correctionId, bankId);
+
+    if (!correction) {
+      throw new NotFoundError(`Failed to find a correction with id '${correctionId}' for bank id '${bankId}'`);
+    }
+
+    // TODO: FN-3688 - validate the form data has the fields expected for the given reasons
+    const validatedFormData = formData as unknown as RecordCorrectionTransientFormData;
```

Comment:
can this line be removed but keep the todo comment.  Should not be adding unknown and this line will be added soon anyways.  same with any other unknowns in this PR

Reply:
Without this line I get a ts error and this PR would not be mergable because actually in here we need a function that parses the data. This is fine to merge because this endpoint isn't being called anywhere so no data will get created on a deployed env even on dev where the feature flag is on until the work is done to replace this type assertion with a parsing function